internal/world/azure.go

```diff
@@ -75,25 +76,26 @@ func (l *LeveledLogrus) fields(keysAndValues ...interface{}) map[string]interfac
 	return fields
 }
 
-func (l *LeveledLogrus) Error(msg string, keysAndValues ...interface{}) {
-	l.WithFields(l.fields(keysAndValues...)).Error(msg)
+func (l *LeveledZerolog) Error(msg string, keysAndValues ...interface{}) {
+	l.logger.Error().Fields(l.fields(keysAndValues...)).Msg(msg)
 }
 
-func (l *LeveledLogrus) Info(msg string, keysAndValues ...interface{}) {
-	l.WithFields(l.fields(keysAndValues...)).Info(msg)
+func (l *LeveledZerolog) Info(msg string, keysAndValues ...interface{}) {
+	l.logger.Info().Fields(l.fields(keysAndValues...)).Msg(msg)
 }
-func (l *LeveledLogrus) Debug(msg string, keysAndValues ...interface{}) {
-	l.WithFields(l.fields(keysAndValues...)).Debug(msg)
+func (l *LeveledZerolog) Debug(msg string, keysAndValues ...interface{}) {
+	l.logger.Debug().Fields(l.fields(keysAndValues...)).Msg(msg)
 }
 
-func (l *LeveledLogrus) Warn(msg string, keysAndValues ...interface{}) {
-	l.WithFields(l.fields(keysAndValues...)).Warn(msg)
+func (l *LeveledZerolog) Warn(msg string, keysAndValues ...interface{}) {
+	l.logger.Warn().Fields(l.fields(keysAndValues...)).Msg(msg)
 }
 
 func (w *World) Azure() *Azure {
 	if w.azure != nil {
 		return w.azure
 	}
+	logger := zerolog.Ctx(w.ctx)
```

Comment:
The question is if we would like to do something like that. (Not sure if it is running code but I guess you get the intention :) )
```suggestion
	logger := zerolog.Ctx(w.ctx).With().Str("component", "azure").Logger()
```

Reply:
Good point. I'll do that and also within the Vault part :)

Reply:
Done :)